core/src/main/java/org/apache/accumulo/core/clientImpl/ConditionalWriterImpl.java

```diff
@@ -704,11 +704,15 @@ private void invalidateSession(SessionID sessionId, HostAndPort location)
         throw new TimedOutException(Collections.singleton(location.toString()));
       }
 
-      sleepUninterruptibly(sleepTime, MILLISECONDS);
+      // interrupt checked in while condition
+      UtilWaitThread.sleep(sleepTime, MILLISECONDS);
       sleepTime = Math.min(2 * sleepTime, MAX_SLEEP);
 
     }
-
+    if (Thread.currentThread().isInterrupted()) {
+      throw new TimedOutException(
+          "Interrupted during sleep: " + Collections.singleton(location.toString()));
+    }
```

Comment:
Why not just call Thread.sleep() and throw InterruptedException then? It has the same effect with less code.